dashboard/lib/api/v1/school_autocomplete.rb

```diff
@@ -26,7 +27,8 @@ def self.get_matches(query, limit, use_new_search)
       return [] if query.length < MIN_WORD_LENGTH + 2
       rows = rows.
         where("MATCH(name,city) AGAINST(? IN BOOLEAN MODE)", query).
-        order("MATCH(name,city) AGAINST('#{query}' IN BOOLEAN MODE) DESC, state, city, name")
+        # sanitize_sql_for_order won't be a public method until Rails 5.2
+        order(ActiveRecord::Base.send(:sanitize_sql_for_order, ["MATCH(name,city) AGAINST(? IN BOOLEAN MODE) DESC, state, city, name", query]))
```

Comment:
In this case, instead of calling `sanitize_sql_for_order` here, can't you just
```rb
order('MATCH(name, city) AGAINST(? IN BOOLEAN MODE) DESC, state, city, name', query)
```

Reply:
The ActiveRecord `order` query method does not appear to sanitize parameter inputs:
```
[development] dashboard > orderby_column = 'id'
=> "id"
[development] dashboard > School.where(name: 'Baton Rouge Magnet High School').order("#{orderby_column}")
  School Load (357.0ms)  SELECT `schools`.* FROM `schools` WHERE `schools`.`name` = 'Baton Rouge Magnet High School' ORDER BY id
=> #<ActiveRecord::Relation [#<School id: "220054000347", school_district_id: 2200540, name: "BATON ROUGE MAGNET HIGH SCHOOL", city: "BATON ROUGE", state: "LA", zip: "70808", school_type: "public", created_at: "2018-03-27 20:04:39", updated_at: "2018-03-27 20:40:40", address_line1: "1105 LEE DRIVE", address_line2: nil, address_line3: nil, latitude: 0.3040574e2, longitude: -0.91152311e2, state_school_id: "LA-017-017008">]>
[development] dashboard > School.where(name: 'Baton Rouge Magnet High School').order('?', orderby_column)
  School Load (2.8ms)  SELECT `schools`.* FROM `schools` WHERE `schools`.`name` = 'Baton Rouge Magnet High School' ORDER BY ?, id
  School Load (2.4ms)  SELECT `schools`.* FROM `schools` WHERE `schools`.`name` = 'Baton Rouge Magnet High School' ORDER BY ?, id
Traceback (most recent call last):
       16: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/bundler/gems/seamless_database_pool-c9a5c2bc92ef/lib/active_record/connection_adapters/seamless_database_pool_adapter.rb:371:in `rescue in proxy_connection_method'
       15: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/bundler/gems/seamless_database_pool-c9a5c2bc92ef/lib/active_record/connection_adapters/seamless_database_pool_adapter.rb:361:in `proxy_connection_method'
       14: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activerecord-5.0.7.2/lib/active_record/connection_adapters/abstract/database_statements.rb:373:in `select'
       13: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activerecord-5.0.7.2/lib/active_record/connection_adapters/mysql/database_statements.rb:36:in `exec_query'
       12: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activerecord-5.0.7.2/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:225:in `execute_and_free'
       11: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activerecord-5.0.7.2/lib/active_record/connection_adapters/mysql/database_statements.rb:31:in `execute'
       10: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activerecord-5.0.7.2/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:218:in `execute'
        9: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activerecord-5.0.7.2/lib/active_record/connection_adapters/abstract_adapter.rb:583:in `log'
        8: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activesupport-5.0.7.2/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
        7: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activerecord-5.0.7.2/lib/active_record/connection_adapters/abstract_adapter.rb:590:in `block in log'
        6: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/activerecord-5.0.7.2/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:218:in `block in execute'
        5: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/rack-mini-profiler-1.0.0/lib/patches/db/mysql2.rb:22:in `query'
        4: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/mysql2-0.5.2/lib/mysql2/client.rb:130:in `query'
        3: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/mysql2-0.5.2/lib/mysql2/client.rb:130:in `handle_interrupt'
        2: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/mysql2-0.5.2/lib/mysql2/client.rb:131:in `block in query'
        1: from /Users/suresh/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/mysql2-0.5.2/lib/mysql2/client.rb:131:in `_query'
ActiveRecord::StatementInvalid (Mysql2::Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '?, id' at line 1: SELECT `schools`.* FROM `schools` WHERE `schools`.`name` = 'Baton Rouge Magnet High School' ORDER BY ?, id)
```

Reply:
Wow... TIL.